crates/rspack_plugin_javascript/src/visitors/mod.rs

```diff
@@ -55,108 +198,44 @@ pub fn run_before_pass(
       assumptions.set_public_class_fields = true;
     }
 
-    let resource_path = resource_data.resource_path.to_string_lossy();
-
     let mut pass = chain!(
       swc_visitor::resolver(unresolved_mark, top_level_mark, syntax.typescript()),
-      //      swc_visitor::lint(
-      //        &ast,
-      //        top_level_mark,
-      //        unresolved_mark,
-      //        EsVersion::Es2022,
-      //        &cm
-      //      ),
       Optional::new(
         swc_visitor::decorator(assumptions, &options.builtins.decorator),
-        syntax.decorators()
+        // Decorator transformation varies from `ModuleType`,
+        // - TypeScript-like: decorators will be transformed by default, with legacy settings and will emit meta data.
+        //      Since this is a default behavior with tsc.
+        // - JavaScript-like: decorators will not be transformed(if `disableTransformByDefault` is on), and the parse will fail.
+        (options.should_transform_by_default() || syntax.typescript()) && syntax.decorators()
```

Comment:
Checkpoint: with `disableTransformByDefault` enabled, this will never visited as it will result in a `ModuleParseError` in the previous parsing stage.